src/interpret.jl

```diff
@@ -417,42 +419,42 @@ function _step_expr!(stack, frame, @nospecialize(node), pc::JuliaProgramCounter,
                     end
                 elseif node.head == :thunk
                     newframe = prepare_thunk(moduleof(frame), node)
-                    frame.pc[] = pc
-                    if isa(stack, Compiled)
-                        finish!(stack, newframe, true)
+                    if isa(recurse, Compiled)
+                        finish!(recurse, newframe, true)
                     else
-                        push!(stack, frame)
-                        finish!(stack, newframe, true)
-                        pop!(stack)
-                        push!(junk, newframe)  # rather than going through GC, just re-use it
+                        newframe.caller = frame
+                        frame.callee = newframe
+                        finish!(recurse, newframe, true)
```

Comment:
Is there a need to set `frame.callee` here in case `finish!` throws?

Reply:
It is set on the line above that

Reply:
I meant a need to set it to `nothing`. But perhaps you made `handle_err` handle that now.

Reply:
I see. I *think* it's taken care of in `handle_err`. I think we should leave that `@assert` in place in `step_expr!`, though, just to be sure.